config/importmap.rb

```diff
@@ -1,9 +1,10 @@
 # frozen_string_literal: true
 
 # Pin npm packages by running ./bin/importmap
-
 pin 'application', preload: true
 pin '@hotwired/turbo-rails', to: 'turbo.min.js', preload: true
 pin '@hotwired/stimulus', to: 'stimulus.min.js', preload: true
 pin '@hotwired/stimulus-loading', to: 'stimulus-loading.js', preload: true
 pin_all_from 'app/javascript/controllers', under: 'controllers'
+pin 'bootstrap', to: 'bootstrap.min.js', preload: true
+pin '@popperjs/core', to: 'popper.js', preload: true
```

Comment:
why is this line necessary?

Reply:
Bootstrap tooltips and popovers depend on the popper.js file for positioning, also the bootstrap gem already depends on the popper_js gem, so these tools must be pinned to the importmap file in the config path